src/Storages/MergeTree/MergeTreeIndexAggregatorBloomFilter.cpp

```diff
@@ -48,17 +48,18 @@ void MergeTreeIndexAggregatorBloomFilter::update(const Block & block, size_t * p
     Block granule_index_block;
     size_t max_read_rows = std::min(block.rows() - *pos, limit);
 
-    for (const auto & index_column_name : index_columns_name)
+    for (size_t column = 0; column < index_columns_name.size(); ++column)
     {
-        const auto & column_and_type = block.getByName(index_column_name);
+        const auto & column_and_type = block.getByName(index_columns_name[column]);
         auto index_column = BloomFilterHash::hashWithColumn(column_and_type.type, column_and_type.column, *pos, max_read_rows);
 
-        granule_index_block.insert({index_column, std::make_shared<DataTypeUInt64>(), column_and_type.name});
+        const auto &index_data = typeid_cast<const ColumnUInt64 *>(index_column.get())->getData();
```

Comment:
`nit:` Instead of doing a `typeid_cast` here, there's already a `checkAndGetColumn` func that can be used directly. It essentially does the same and you can probably use it in the following way:
```suggestion
        const auto &index_col = checkAndGetColumn<ColumnUInt64>(index_column.get());
        const auto & index_data = index_col->getData();
```